src/check-label/check-label.ts

```diff
@@ -0,0 +1,88 @@
+// check-label/check-label.ts
+
+import process from 'node:process';
+import path from 'node:path';
+import assert from 'node:assert';
+import { readFile } from 'node:fs/promises';
+import { debug } from 'debug';
+import { setFailed } from '@actions/core';
+import semver from 'semver';
+
+import { getFileFromMain, getLabelsOnPR } from '../github-api';
+
+const log = debug('check-label');
+
+interface PackageJSON {
+  name: string;
+  version: string;
+  files: string[];
+}
+
+async function getLocalPackageJsonVersion(fileName: string): Promise<string> {
+  const packageJSONPath = path.join(process.cwd(), fileName);
+  const readPackageJson = await readFile(packageJSONPath, 'utf8');
+  const packageJson = JSON.parse(readPackageJson) as PackageJSON;
+
+  return packageJson.version;
+}
+
+export function validateVersionCompareMatchesSemver(
+  branchPackageJsonVersion: string,
+  mainPackageJsonVersion: string
+): semver.ReleaseType | null {
+  const semVersionDiff = semver.diff(branchPackageJsonVersion, mainPackageJsonVersion);
+  const newVersionList = branchPackageJsonVersion.split('.');
+
+  if (semVersionDiff === 'minor' && Number(newVersionList[2]) !== 0) {
+    throw new Error('Minor version bump but patch version is not 0');
+  }
+
+  if (semVersionDiff === 'major' && (Number(newVersionList[1]) !== 0 || Number(newVersionList[2]) !== 0)) {
+    throw new Error('Major version bump but minor and patch version is not 0');
+  }
+
+  return semVersionDiff;
+}
+
+export default async function (): Promise<void> {
+  log('Action start');
+
+  const labelsPullRequest = await getLabelsOnPR();
+  if (labelsPullRequest.length > 1) {
+    throw new Error('PR has more than one label');
+  }
+  const label = labelsPullRequest[0]?.toLowerCase();
+  assert(label, 'Unable to get label from PR');
+
+  const branchPackageJsonVersion = await getLocalPackageJsonVersion('package.json');
+  const mainPackageJsonVersionRaw = await getFileFromMain('package.json');
+
+  if (!mainPackageJsonVersionRaw) {
+    throw new Error('Unable to get package.json from main branch');
+  }
+  const mainPackageJsonVersion = JSON.parse(mainPackageJsonVersionRaw) as PackageJSON;
+
+  const packageVersionDiff = validateVersionCompareMatchesSemver(
```

Comment:
unless I'm reading this wrong, I think there's still an issue.  e.g. you can go from 1.2.3 to 8.0.0?   I think the logic should be given the existing package.json version, and a label, what should the new version be.  and if that doesn't equal the branch version, fail.

Reply:
Yup I've gone with that approach.